### PR TITLE
add CreateEventSchema button

### PR DESCRIPTION
**Pull request by jean-neiverth**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +22,7 @@
     "react": "^18",
     "react-dom": "^18",
     "react-query": "^3.39.3",
+    "styled-components": "^6.1.12",
```
Comment (ribeirojose) [inline package.json:25]: 🤔 why use styled components when we have tailwind in the project?

File: src/components/CreateEventSchemaButton.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+// const newEvent = {
+//     owner:"0xa90914762709441d557De208bAcE1edB1A3968b2",
+//     name:"test",
+//     briefDescription:"test description",
+//     fullDescription:"Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptate, numquam! Esse officia molestiae illum nihil dolorum rem debitis laborum ab. Quaerat voluptatibus repellendus deserunt porro distinctio laboriosam earum nesciunt laborum?",
+//     startsAt: today.valueOf(),
+//     endsAt: tomorrow.valueOf(),
+//     imageUrl:"https://picsum.photos/id/237/200/300",
+// } as EventSchema
+// console.log('newEvent', newEvent)
```
Comment (ribeirojose) [inline src/components/CreateEventSchemaButton.tsx:16]: 👁️

File: src/wagmi.ts [hunk 3]
```diff
@@ -1,10 +1,10 @@
 import { http, cookieStorage, createConfig, createStorage } from "wagmi";
-import { mainnet, optimism, sepolia } from "wagmi/chains";
+import { mainnet, optimism, sepolia, optimismSepolia } from "wagmi/chains";
```
Comment (ribeirojose) [inline src/wagmi.ts]: we can probably forgo mainnet and sepolia altogether here, no need to support them

File: src/hooks/useSigner.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+// based on https://gist.github.com/slavik0329/2e5b6fc31cb745b65d3d37f7cf1d7b36
```
Comment (ribeirojose) [inline src/hooks/useSigner.ts:1]: 👍

File: src/components/CreateEventSchemaButton.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    const registrySchema = useRegistrySchema();
+
+    async function handleCreateEventSchema() {
+
+        registrySchema({
+            schema:CREATE_EVENT_SCHEMA,
+            resolver:zeroAddress,
+            revocable:true,
+        })
+    }
```
Comment (ribeirojose) [inline src/components/CreateEventSchemaButton.tsx]: no need for thus to be defined within the component. But is this definition needed at all?

File: src/components/CreateEventSchemaButton.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+const Button = styled.button`
+    padding: 0.5rem 1rem;
+    background-color: gray;
+    margin-right:1rem;
+    border-radius:10px;
+
+    &:hover {
+        background-color: #4a4a4a;
+    }
+`
```
Comment (ribeirojose) [inline src/components/CreateEventSchemaButton.tsx]: `#could` use tailwind to define css instead

File: src/hooks/useRegistrySchema.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
+import { useSigner } from "@/hooks/useSigner";
+import { encodePacked, keccak256, zeroAddress } from "viem";
+import { EAS, SchemaRegistry } from "@ethereum-attestation-service/eas-sdk";
+import { useCallback, useEffect, useState } from "react";
+
+interface RegistrySchemaParms {
+    schema: string,
+    resolver:`0x${string}`,
```
Comment (ribeirojose) [inline src/hooks/useRegistrySchema.ts]: `#could` use address from viem

File: src/hooks/useRegistrySchema.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
+import { useSigner } from "@/hooks/useSigner";
+import { encodePacked, keccak256, zeroAddress } from "viem";
+import { EAS, SchemaRegistry } from "@ethereum-attestation-service/eas-sdk";
+import { useCallback, useEffect, useState } from "react";
+
+interface RegistrySchemaParms {
+    schema: string,
```
Comment (ribeirojose) [inline src/hooks/useRegistrySchema.ts:8]: this is also a `0x${string}` right?

File: src/hooks/useRegistrySchema.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    resolver:`0x${string}`,
+    revocable:boolean,
```
Comment (ribeirojose) [inline src/hooks/useRegistrySchema.ts:10]: please format/lint your files when opening the PR!

File: src/hooks/useRegistrySchema.ts [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+};
+
+export const useRegistrySchema = () => {
+
+    // Optimism Sepolia Version 1.0.2
```
Comment (ribeirojose) [inline src/hooks/useRegistrySchema.ts]: is this a relevant comment?

File: src/hooks/useRegistrySchema.ts [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+    const EAS_CONTRACT_ADDRESS = "0x4200000000000000000000000000000000000021" //Deployment and ABI: EAS.json
+    const SCHEMA_REGISTRY_CONTRACT_ADDRESS = "0x4200000000000000000000000000000000000020" //Deployment and ABI: SchemaRegistry.json
```
Comment (ribeirojose) [inline src/hooks/useRegistrySchema.ts]: are these relevant comments?

File: src/components/Header.tsx [hunk 12]
```diff
@@ -1,91 +1,8 @@
-// "use client";
```
Comment (luizakp) [inline src/components/Header.tsx:1]: 👍

File: src/hooks/useSigner.ts (hunk 4 above)
Comment (luizakp) [inline src/hooks/useSigner.ts:1]: 🔝

File: package.json (hunk 1 above)
Comment (luizakp) [reply package.json:25]: also not being used, `#should` remove it

**Review by luizakp — COMMENTED** [review]
👍 

note: try to address or react or reply to the reviews before merge it. Me as a reviewer understand that you ignored it 😅. Communication always!

**Pull request merged**